Tests/UnitTests/Core/Seeding/BinnedGroupTest.cpp

```diff
@@ -0,0 +1,256 @@
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Seeding/BinnedGroup.hpp"
+#include "Acts/Utilities/Grid.hpp"
+#include "Acts/Utilities/GridBinFinder.hpp"
+#include "Acts/Utilities/GridIterator.hpp"
+
+#include <array>
+#include <memory>
+#include <vector>
+
+namespace Acts::Test {
+
+BOOST_AUTO_TEST_CASE(binned_group_constructor) {
+  Acts::detail::EquidistantAxis xAxis(0, 100, 10);
+  Acts::detail::EquidistantAxis yAxis(0, 100, 10);
+  Acts::detail::EquidistantAxis zAxis(0, 100, 10);
+
+  using grid_1d_t =
+      Acts::Grid<std::vector<std::size_t>, Acts::detail::EquidistantAxis>;
+  using grid_2d_t =
+      Acts::Grid<std::vector<std::size_t>, Acts::detail::EquidistantAxis,
+                 Acts::detail::EquidistantAxis>;
+  using grid_3d_t =
+      Acts::Grid<std::vector<std::size_t>, Acts::detail::EquidistantAxis,
+                 Acts::detail::EquidistantAxis, Acts::detail::EquidistantAxis>;
+
+  grid_1d_t grid_1d(std::make_tuple(xAxis));
+  grid_2d_t grid_2d(std::make_tuple(xAxis, yAxis));
+  grid_3d_t grid_3d(std::make_tuple(xAxis, yAxis, zAxis));
+
+  std::unique_ptr<Acts::GridBinFinder<1ul>> binFinder_1d =
+      std::make_unique<Acts::GridBinFinder<1ul>>(1);
+  std::unique_ptr<Acts::GridBinFinder<2ul>> binFinder_2d =
+      std::make_unique<Acts::GridBinFinder<2ul>>(1, 1);
+  std::unique_ptr<Acts::GridBinFinder<3ul>> binFinder_3d =
+      std::make_unique<Acts::GridBinFinder<3ul>>(1, 2, 1);
+
+  std::array<std::vector<std::size_t>, 1ul> navigation_1d;
+  navigation_1d[0ul].resize(10);
+  std::array<std::vector<std::size_t>, 2ul> navigation_2d;
+  navigation_2d[1ul].resize(10);
+
+  // Costructors
+  // We provide a proper navigation
+  Acts::BinnedGroup<grid_1d_t> group_1d(std::move(grid_1d), *binFinder_1d.get(),
+                                        *binFinder_1d.get(),
+                                        std::move(navigation_1d));
+  // We provide a partial navigation, the constructor will complete it
+  Acts::BinnedGroup<grid_2d_t> group_2d(std::move(grid_2d), *binFinder_2d.get(),
+                                        *binFinder_2d.get(),
+                                        std::move(navigation_2d));
+  // We do not provide navigation, the constructor will define it
+  Acts::BinnedGroup<grid_3d_t> group_3d(std::move(grid_3d), *binFinder_3d.get(),
+                                        *binFinder_3d.get());
+
+  // Move Constructor/Assignment
+  const Acts::BinnedGroup<grid_1d_t> group_1d_moved(std::move(group_1d));
+  const Acts::BinnedGroup<grid_2d_t> group_2d_moved(std::move(group_2d));
+  const Acts::BinnedGroup<grid_3d_t> group_3d_moved = std::move(group_3d);
+
+  [[maybe_unused]] const grid_1d_t& retrievedGrid = group_1d.grid();
```

Comment:
Do you need the `maybe_unused` here?

Reply:
not sure. Wouldn't you have compilation warnings if we don't use it?